common/predictive-text/worker/model-compositor.ts

```diff
@@ -24,15 +48,46 @@ class ModelCompositor {
           pair.sample.transformId = transform.id;
         }
 
-        let compositedPair = {sample: pair.sample, p: pair.p * alt.p};
-        suggestionDistribution.push(compositedPair);
+        // Combine duplicate samples.
+        let displayText = pair.sample.displayAs;
+
+        if(displayText == keepOptionText) {
+          keepOption = pair.sample;
+          keepOption.isKeep = true;
+        } else {
+          let s = suggestionDistribMap[displayText];
```

Comment:
What is `s`? Could you please rename this? An individual suggestion? I'm a bit unclear as to what this branch does.